src/modes/ModeEstimation.jl

```diff
@@ -37,45 +53,38 @@ The `OptimizationContext` transforms variables to their constrained space, but
 does not use the density with respect to the transformation. This context is
 intended to allow an optimizer to sample in R^n freely.
 """
-struct OptimizationContext{C<:AbstractContext} <: AbstractContext
-    context::C
-end
-
-DynamicPPL.NodeTrait(::OptimizationContext) = DynamicPPL.IsParent()
-DynamicPPL.childcontext(context::OptimizationContext) = context.context
-DynamicPPL.setchildcontext(::OptimizationContext, child) = OptimizationContext(child)
+struct DefaultOptimizationContext <: AbstractContext end
+DynamicPPL.NodeTrait(::DefaultOptimizationContext) = DynamicPPL.IsLeaf()
 
 # assume
-function DynamicPPL.tilde_assume(ctx::OptimizationContext{<:LikelihoodContext}, dist, vn, vi)
-    r = vi[vn, dist]
-    return r, 0, vi
-end
-
-function DynamicPPL.tilde_assume(ctx::OptimizationContext, dist, vn, vi)
+function DynamicPPL.tilde_assume(ctx::DefaultOptimizationContext, dist, vn, vi)
     r = vi[vn, dist]
     return r, Distributions.logpdf(dist, r), vi
 end
 
 # dot assume
-function DynamicPPL.dot_tilde_assume(ctx::OptimizationContext{<:LikelihoodContext}, right, left, vns, vi)
-    # Values should be set and we're using `SampleFromPrior`, hence the `rng` argument shouldn't
-    # affect anything.
-    # TODO: Stop using `get_and_set_val!`.
-    r = DynamicPPL.get_and_set_val!(Random.default_rng(), vi, vns, right, SampleFromPrior())
-    return r, 0, vi
-end
-
 _loglikelihood(dist::Distribution, x) = loglikelihood(dist, x)
 _loglikelihood(dists::AbstractArray{<:Distribution}, x) = loglikelihood(arraydist(dists), x)
 
-function DynamicPPL.dot_tilde_assume(ctx::OptimizationContext, right, left, vns, vi)
+function DynamicPPL.dot_tilde_assume(ctx::DefaultOptimizationContext, right, left, vns, vi)
     # Values should be set and we're using `SampleFromPrior`, hence the `rng` argument shouldn't
     # affect anything.
     # TODO: Stop using `get_and_set_val!`.
     r = DynamicPPL.get_and_set_val!(Random.default_rng(), vi, vns, right, SampleFromPrior())
     return r, _loglikelihood(right, r), vi
 end
 
+# Shared.
+const OptimizationContext = Union{LikelihoodOptimizationContext,DefaultOptimizationContext}
```

Comment:
This means it's a breaking change.

Reply:
But `OptimizationContext` isnn't being exported? 

It's exported from the module, but not from Turing.

Reply:
Ah, good point, I missed that 👍